### PR TITLE
drivers: sdhc: add support for using CPOL/CPHA SPI clock modes

### PR DESCRIPTION
**Pull request by LeHack**
In order to use a simple SPI-driven MH-SD Card Module with STM's Nucleo-F429ZI, I found it necessary to change the SPI's clock polarity and phase during the SD card initialization (despite some internet sources arguing that it shouldn't be required).
Hence I decided to propose adding these as devicetree-level options for the sdhc-spi driver (as well as sample configuration).

### CODE REVIEW

**Comment by danieldegrasse** [comment]
@LeHack changes look good to me, but please address merge conflict

**Comment by LeHack** [comment]
Conflict resolved, checks green, just waiting for approval.

**Review by danieldegrasse — APPROVED** [review]

**Comment by LeHack** [comment]
@nashif @galak Any chance you could have a look at this?

**Review by de-nordic — APPROVED** [review]

**Pull request merged**
